### DISCUSSION TITLE
How to add clearing functionality to a property value

### DISCUSSION THREAD

**Discussion by vamshi0** [GrapesJS/grapesjs#4626] [Q&A]
Hi friends,

How to make a property to clear its value ?

Here in the image example, how can i add clear button(x) to each individual property
<img width="137" alt="Screenshot 2022-09-28 143108" src="https://user-images.githubusercontent.com/19800624/192738513-4524dcab-a732-4f08-9163-03232c91cc73.png">

i have gone through documentation, but i am not getting the property.canClear() to work.

how to get the correct property instance and set the clear property ?

Any help will be appreciated, Thank you..!

**Comment by artf** [comment]
That's the default UI for composite types and it doesn't make sense to have clear for each input as it acts like `padding: 1px 2px...` (so it makes sense here only to clear the whole property). If you want you can always create your own Style Manager with your input components and logic https://grapesjs.com/docs/modules/Style-manager.html#customization
